NDTensors/src/UnallocatedArrays/src/unallocated_impl.jl

```diff
@@ -0,0 +1,27 @@
+for Typ in (:UnallocatedFill, :UnallocatedZeros)
+  ## Here are functions specifically defined for UnallocatedArrays
+  ## not implemented by FillArrays
+  ## TODO use set_parameters/get_parameters instead of alloctype and other 
+  ## type info functions.
+  ## TODO determine min number of functions needed to be forwarded
+  @eval begin
+    alloctype(A::$Typ) = alloctype(typeof(A))
+    alloctype(Atype::Type{<:$Typ}) = get_parameter(Atype, Position{4}())
+
+    Array(A::$Typ) = alloctype(typeof(A))
```

Comment:
This definition doesn't make sense to me, it seems to be outputing a type but I would expect `Array` to output an allocated `Array` object.

Reply:
This actually does return an allocated Array object
```
julia> z = Fill{Float32}(3, (2,3))
julia> Z = UnallocatedFill{Float32, 2, typeof(axes(z)), Matrix{Float32}}(z)
julia> Array(Z)
  2×3 Matrix{Float32}:
 3.0  3.0  3.0
 3.0  3.0  3.0
```

Reply:
Hmm so then I'm confused what `alloctype` is doing, since I expect functions with the name `*type(...)` to output a type, not an object.

Reply:
Also it doesn't make sense to call this `Array`, since it seems like it would output a `CuArray` if that was the allocation type, but a function with the name `Array` should always output an `Array`. Seems like what we want is a function called `alloc` or `allocate`.

Reply:
Okay I will modify this to by allocate and have it defined as
```
allocate(A::$Typ) = alloctype(A)(parent(A))
```
and remove Array and in my call above my implementation wasn't being called, it was using `Base.Array` so it makes sense to remove it

Reply:
Sounds like a good plan.